### DISCUSSION TITLE
How to add static font files as a fresh plugin

### DISCUSSION THREAD

**Discussion by CarcajadaArtificial** [freshframework/fresh#2391] [Q&A]
Hey there! I'm currently developing a component library for Fresh. Right now, I'm working on setting up the plugin aspect of the library. I've managed to configure all the static CSS and islands successfully. However, I'm exploring options (if feasible) to include static font files (.woff2 and similar) without requiring users of my library to manually add them to the /static directory and import them in the /routes/_app.tsx function.

I asked around on the official Discord and the Deno AI Helper Bot answered that it is possible using the `staticAssets` attribute but it also warned me about unavailable documentation for said attribute and suggested that I should experiment a bit. So I did, and I con't find any way for it to work so I believe the bot was completely hallucinating?

Appreciate any guidance! 😄

**Comment by marvinhagemeister** [comment]
The only way to do this without relying on Fresh's `static/` directory would be by writing a custom middleware and serve the files through that yourself.

**Reply by CarcajadaArtificial** [reply]
Hey! Thanks for the answer. I'll look into `buildStart` and `ResolvedFreshConfig` to make this implementation.
